### ISSUE TITLE
ArrayBroadcastChannel throwing NPE for low capacity

### ISSUE THREAD

**Issue by bj0** [Kotlin/kotlinx.coroutines#138]
I was using the following example code to test `BroadcastChannel` and how it would handle different speed consumers and noticed that if I set the capacity to 1 or 2,  the consumers immediately throw `NullPointerException`

```
fun main(args: Array<String>) = runBlocking {
    val chan = BroadcastChannel<Int>(2)

    launch {
        for (i in 0 until 1000) {
            println("push $i")
            chan.send(i)
            delay(1000)
        }
    }

    launch {
        val sub = chan.openSubscription()
        println("start job1")
        for (i in sub) {
            println("job1 got $i")
            delay(1000)
        }
    }

    launch {
        val sub = chan.openSubscription()
        println("start job2")
        for (i in sub) {
            println("job2 got $i")
            delay(3000)
        }
    }

    delay(10000)
    println("done")
}
```
push 0
start job1
start job2

```
Exception in thread "ForkJoinPool.commonPool-worker-2" Exception in thread "ForkJoinPool.commonPool-worker-3" java.lang.NullPointerException
	at TestBroadcastChannelKt$main$1$2.doResume(TestBroadcastChannel.kt:21)
	at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resume(CoroutineImpl.kt:54)
	at kotlinx.coroutines.experimental.DispatchTask.run(CoroutineDispatcher.kt:122)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
java.lang.NullPointerException
	at TestBroadcastChannelKt$main$1$3.doResume(TestBroadcastChannel.kt:30)
	at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resume(CoroutineImpl.kt:54)
	at kotlinx.coroutines.experimental.DispatchTask.run(CoroutineDispatcher.kt:122)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```
The lines in the stacktrace correspond with the `delay` calls

**Comment by elizarov** [comment]
Thanks for the report. That is definitely a bug somewhere.

**Comment by elizarov** [comment]
Released in version 0.19.1
